src/main/java/com/gmail/goosius/siegewar/command/SiegeWarAdminCommand.java

```diff
@@ -207,12 +207,12 @@ private void parseSiegeWarImmunityCommand(CommandSender sender, String[] args) {
 			showImmunityHelp(sender);
 			return;
 		}
-
 		try {
-			if (args[0].equalsIgnoreCase("alltowns"))
+			if (args[0].equalsIgnoreCase("alltowns") && !args[1].equalsIgnoreCase("permanent")) {
 				Integer.parseInt(args[1]);
-			else
+			} else if (!args[2].equalsIgnoreCase("permanent")){
 				Integer.parseInt(args[2]);
+			}
 		} catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
 			Messaging.sendMsg(sender, Translation.of("msg_error_must_be_num"));

```

Comment:
- The above translation string is currently this: '&cAmount must be a number.'`
- Along with the new permanent option, this text should be changed to suit .e.g. '&cAmount must be a number of permanent'

Reply:
This language string still needs updating
- e.g. "msg_error_must_be_num" needs to be changed
- I suggest removing the current string and adding a new one right at the end of the file called: `msg_error_must_be_num_or_permanent`
- Make sure to change this in both the english.yml and french.yml files

Reply:
Can do, but how should I go about getting a french translation? Just use google translate or is there someone I should message?

Reply:
To update the french.yml file
1. Create a new comment at the bottom of the file called `#Added in 23` 
2. Add the new lines there, just like in the english.yml file 
3. (No need to translate them yourself)